src/asyncresults.jl

```diff
@@ -0,0 +1,256 @@
+"An asynchronous PostgreSQL query"
+mutable struct AsyncResult
+    "The LibPQ.jl Connection used for the query"
+    jl_conn::Connection
+
+    "Keyword arguments to pass to Result on creation"
+    result_kwargs::Ref
+
+    "Whether or not the query should be cancelled, if running"
+    should_cancel::Bool
+
+    "Task which errors or returns a LibPQ.jl Result which is created once available"
+    result_task::Task
+
+    function AsyncResult(jl_conn::Connection, result_kwargs::Ref)
+        async_result = new(jl_conn, result_kwargs, false)
+        jl_conn.async_result = async_result
+        return async_result
+    end
+end
+
+function AsyncResult(jl_conn::Connection; kwargs...)
+    return AsyncResult(jl_conn, Ref(kwargs))
+end
+
+function Base.show(io::IO, async_result::AsyncResult)
+    status = if isready(async_result)
+        if iserror(async_result)
+            "errored"
+        else
+            "finished"
+        end
+    else
+        "in progress"
+    end
+    print(io, typeof(async_result), " (", status, ")")
+end
```

Comment:
This `show` is parsable, but `eval(Meta.parse(repr(result))) != result`.
Which is probably fine, since it is an error.

Reply:
I don't think it is, as the space before the open parenthesis should cause a syntax error.

Reply:
Ah good